cmd/stack_ls.go

```diff
@@ -75,93 +73,77 @@ func newStackLsCmd() *cobra.Command {
-					maxname = len(name)
+			maxName := 47
+			for _, summary := range stackSummaries {
+				name := summary.Name().String()
+				if len(name) > maxName {
+					maxName = len(name)
 				}
 			}
+			maxName++ // Account for adding the '*' to the currently selected stack.
 
-			// We have to fault in snapshots for all the stacks we are going to list here, because that's the easiest
-			// way to get the last update time and the resource count.  Since this is an expensive operation, we'll
-			// do it before printing any output so the latency happens all at once instead of line by line.
-			//
-			// TODO[pulumi/pulumi-service#1530]: We need a lighterweight way of fetching just the specific information
-			// we want to display here.
-			for _, name := range stackNames {
-				stack := stacks[name]
-				_, err := stack.Snapshot(commandContext())
-				contract.IgnoreError(err) // If we couldn't get snapshot for the stack don't fail the overall listing.
-			}
-
-			formatDirective := "%-" + strconv.Itoa(maxname) + "s %-24s %-18s"
+			// Header string and formatting options to align columns.
+			formatDirective := "%-" + strconv.Itoa(maxName) + "s %-24s %-18s"
 			headers := []interface{}{"NAME", "LAST UPDATE", "RESOURCE COUNT"}
-
 			if showURLColumn {
 				formatDirective += " %s"
 				headers = append(headers, "URL")
 			}
-
 			formatDirective = formatDirective + "\n"
 
 			fmt.Printf(formatDirective, headers...)
-			for _, name := range stackNames {
-				// Mark the name as current '*' if we've selected it.
-				stack := stacks[name]
+			for _, summary := range stackSummaries {
+				const none = "n/a"
+
+				// Name column
+				name := summary.Name().String()
 				if name == current {
 					name += "*"
 				}
 
-				// Get last deployment info, provided that it exists.
-				none := "n/a"
+				// Last update column
 				lastUpdate := none
-				resourceCount := none
-				snap, err := stack.Snapshot(commandContext())
-				contract.IgnoreError(err) // If we couldn't get snapshot for the stack don't fail the overall listing.
+				if stackLastUpdate := summary.LastUpdate(); stackLastUpdate != nil {
+					lastUpdate = humanize.Time(*stackLastUpdate)
+				}
 
-				if snap != nil {
-					if t := snap.Manifest.Time; !t.IsZero() {
-						lastUpdate = humanize.Time(t)
-					}
-					resourceCount = strconv.Itoa(len(snap.Resources))
+				// ResourceCount column
+				resourceCount := none
+				if stackResourceCount := summary.ResourceCount(); stackResourceCount != nil {
+					resourceCount = strconv.Itoa(*stackResourceCount)
 				}
 
+				// Render the columns.
 				values := []interface{}{name, lastUpdate, resourceCount}
 				if showURLColumn {
 					var url string
-					if cs, ok := stack.(httpstate.Stack); ok {
-						if u, urlErr := cs.ConsoleURL(); urlErr == nil {
-							url = u
+					if httpBackend, ok := b.(httpstate.Backend); ok {
+						if nameSuffix, err := httpBackend.StackConsoleURL(summary.Name()); err != nil {
+							url = "err"
```

Comment:
I would set this to `none` instead of `"err"`, here.

Reply:
Done.